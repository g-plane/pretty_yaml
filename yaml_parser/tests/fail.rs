use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn fmt_snapshot() {
    glob!("fail/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let err_msg = if let Err(err) = yaml_parser::parse(&input) {
            err.to_string()
        } else {
            panic!("expected '{}' to fail", path.display());
        };

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, err_msg);
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
