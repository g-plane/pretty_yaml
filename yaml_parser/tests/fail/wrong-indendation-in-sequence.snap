---
source: yaml_parser/tests/fail.rs
---
parse error at line 4, column 3
  |
4 |   - wrong
  |   ^
invalid indentation
